### PR TITLE
chore(core): removed internal _elgg_config helper function

### PR DESCRIPTION
**Pull request by jdalsem**

### CODE REVIEW

File: engine/classes/ElggRelationship.php [hunk 1]
```diff
@@ -131,7 +131,7 @@ public function getURL() {
 			$subtype = $this->getSubtype();
 
 			$function = "";
-			$handlers = _elgg_config()->relationship_url_handler;
+			$handlers = _elgg_services()->config->relationship_url_handler;
```
Comment (jeabakker) [inline engine/classes/ElggRelationship.php:134]: I believe this section can be removed from `ElggRelationship`

File: engine/tests/phpunit/integration/Elgg/Integration/ElggCoreEntityTest.php [hunk 2]
```diff
@@ -189,7 +189,7 @@ public function testElggEntitySaveAndDelete() {
 	}
 
 	public function testElggEntityDisableAndEnable() {
-		$CONFIG = _elgg_config();
+		$CONFIG = _elgg_services()->config;
```
Comment (jeabakker) [inline engine/tests/phpunit/integration/Elgg/Integration/ElggCoreEntityTest.php:192]: is `$CONFIG` really needed or can this be fixed in a different way (`elgg()->db->prefix`)

File: mod/developers/views/default/admin/developers/settings.php [hunk 3]
```diff
@@ -15,8 +15,8 @@
 	'simple_cache' => [
 		'#type' => 'checkbox',
 		'value' => 1,
-		'checked' => _elgg_config()->simplecache_enabled == 1,
-		'disabled' => _elgg_config()->hasInitialValue('simplecache_enabled'),
+		'checked' => elgg()->config->simplecache_enabled == 1,
```
Comment (jeabakker) [inline mod/developers/views/default/admin/developers/settings.php:18]: why not use `$config` as prepared on line 6

**Review by jeabakker — COMMENTED** [review]
I also wee the pattern `_elgg_services()->config->dbprefix` a lot, should this come from `_elgg_services()->db->dbprefix`?

File: engine/classes/ElggRelationship.php (hunk 1 above)
Comment (jdalsem) [reply engine/classes/ElggRelationship.php:134]: think so too

**Review by jeabakker — APPROVED** [review]

**Pull request merged**